# ****Lab 2 - Machine Learning****

## **Load the dataset**

In [10]:
# Download the SMS Spam Collection dataset from Kaggle
import kagglehub
import os
import pandas as pd

path = kagglehub.dataset_download("uciml/sms-spam-collection-dataset")

print(path)

Using Colab cache for faster access to the 'sms-spam-collection-dataset' dataset.
/kaggle/input/sms-spam-collection-dataset


In [11]:
csv_path = os.path.join(path, "spam.csv")
df = pd.read_csv(csv_path, encoding="latin-1")

## **Display the shape**

In [12]:
df.shape

(5572, 5)

The dataset contains 5,572 rows (SMS messages) and 5 columns. Each row represents one SMS message

## **Preview the first few rows**

In [13]:
df.head()

v1                                                 v2 Unnamed: 2  \
0   ham  Go until jurong point, crazy.. Available only ...        NaN   
1   ham                      Ok lar... Joking wif u oni...        NaN   
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...        NaN   
3   ham  U dun say so early hor... U c already then say...        NaN   
4   ham  Nah I don't think he goes to usf, he lives aro...        NaN   

  Unnamed: 3 Unnamed: 4  
0        NaN        NaN  
1        NaN        NaN  
2        NaN        NaN  
3        NaN        NaN  
4        NaN        NaN

The first few rows show that column v1 contains the class labels (spam or ham), while column v2 contains the SMS message text. The remaining columns contain missing values

## **Check column names and data types**

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5572 entries, 0 to 5571
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   v1          5572 non-null   object
 1   v2          5572 non-null   object
 2   Unnamed: 2  50 non-null     object
 3   Unnamed: 3  12 non-null     object
 4   Unnamed: 4  6 non-null      object
dtypes: object(5)
memory usage: 217.8+ KB


The dataset consists of 5,572 entries. Columns v1 and v2 have no missing values and are of type object. The columns Unnamed: 2, Unnamed: 3, and Unnamed: 4 contain many missing values and are not relevant for the classification task

## **Extra Codes**
After inspecting the dataset, additional preprocessing steps will be applied to improve the data quality

In [15]:
df = df.drop(["Unnamed: 2", "Unnamed: 3", "Unnamed: 4"], axis=1)

In [17]:
df = df.rename(columns={"v1":"label", "v2":"text"})

In [18]:
df.head()

label                                               text
0   ham  Go until jurong point, crazy.. Available only ...
1   ham                      Ok lar... Joking wif u oni...
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...
3   ham  U dun say so early hor... U c already then say...
4   ham  Nah I don't think he goes to usf, he lives aro...

Columns 2,3,4 contain no important data and can be deleted.
Also, we rename column v1 as "label" and v2 as "text"